hivemq-edge/src/frontend/src/components/MQTT/EntityCreatableSelect.tsx

```diff
@@ -0,0 +1,221 @@
+  options: EntityOption[]
+  icon: FC<IconProps>
+  onChange: (newValue: OnChangeValue<string, boolean>) => void
+  value: PropsValue<string>
+}
+
+const EntityCreatableSelect: FC<EntitySelectProps> = ({
+  type,
+  isCreatable = false,
+  isMulti = false,
+  options,
+  isLoading,
+  id,
+  onChange,
+  value,
+}) => {
+  const { t } = useTranslation('components')
+  const safeValue = useMemo(() => {
+    if (value === null) return null
+    if (isMulti) return (value as MultiValue<string>).map((w: string) => ({ label: w, value: w }))
+    return { label: value as string, value: value as string }
+  }, [isMulti, value])
+  const safeOnChange = useCallback(
+    (value: MultiValue<EntityOption> | SingleValue<EntityOption>) => {
+      if (value === null) {
+        onChange(null)
+        return
+      }
+      if (isMulti) {
+        onChange((value as MultiValue<EntityOption>).map((w) => w.value))
+        return
+      }
+
+      onChange((value as EntityOption).value)
+    },
+    [isMulti, onChange]
+  )
+
+  const filterConfig = {
+    trim: false,
+  }
+
+  const SelectComponent = isCreatable ? CreatableSelect : Select
+
+  return (
+    <SelectComponent<EntityOption, boolean, GroupBase<EntityOption>>
+      options={options}
+      isMulti={isMulti}
+      isLoading={isLoading}
+      aria-label={t('EntityCreatableSelect.aria-label', { context: type })}
+      placeholder={t('EntityCreatableSelect.placeholder', { context: type })}
+      noOptionsMessage={() => t('EntityCreatableSelect.options.noOptionsMessage', { context: type })}
+      formatCreateLabel={(entity) => t('EntityCreatableSelect.options.createLabel', { context: type, entity: entity })}
+      id={id}
+      instanceId={type}
+      inputId={`react-select-${type}-input`}
+      isClearable
+      isSearchable
+      selectedOptionStyle="check"
+      components={customComponents(isMulti, type)}
+      filterOption={createFilter(filterConfig)}
+      value={safeValue}
+      onChange={safeOnChange}
+    />
+  )
+}
+
+export default EntityCreatableSelect
+
+interface CreateSelectableProps {
+  id?: string
+  isCreatable?: boolean
+  isMulti?: boolean
+  onChange: (newValue: OnChangeValue<string, boolean>) => void
+  value: PropsValue<string>
+}
+
+export const SelectTopic: FC<CreateSelectableProps> = (props) => {
+  const { data, isLoading } = useGetEdgeTopics({ publishOnly: false })
+  const options =
+    data.map<EntityOption>((e) => ({
+      label: e,
+      value: e,
+    })) || []
+
+  return (
+    <EntityCreatableSelect
+      type={SelectEntityType.TOPIC}
+      options={options}
+      isLoading={isLoading}
+      icon={TopicIcon}
+      {...props}
+    />
+  )
+}
+
+export const SelectTopicFilter: FC<CreateSelectableProps> = (props) => {
+  const { data, isLoading } = useListTopicFilters()
+  const options =
+    data?.items?.map<EntityOption>((e) => ({
+      label: e.topicFilter,
+      value: e.topicFilter,
+      description: e.description,
+    })) || []
```

Comment:
question: I believe that is a `topic`

```suggestion
    data?.items?.map<EntityOption>((topic) => ({
      label: topic.topicFilter,
      value: topic.topicFilter,
      description: topic.description,
    })) || []
```